ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1172,10 +1250,15 @@ void maybeNotifyForChunkCreation(PutChunk chunk) {
      * Notify for the creation of the first chunk. To be called after the overall operation is completed if the overall
      * blob is composite. If no first chunk was put successfully, this will do nothing.
      */
-    void notifyForFirstChunkCreation() {
-      String chunkId = firstChunkIdAndProperties.getFirst().getID();
-      BlobProperties chunkProperties = firstChunkIdAndProperties.getSecond();
-      notificationSystem.onBlobCreated(chunkId, chunkProperties, userMetadata, NotificationBlobType.DataChunk);
+    void maybeNotifyForFirstChunkCreation() {
+      if (indexToChunkIds.get(0) != null) {
+        // reason to check for not null: there are chances that 2nd chunk would completes before the first chunk and
```

Comment:
rather than change this, can you make the change in the `onComplete()` of the `PutManager` to only call `op.maybeNotifyForBlobCreation()` in the successful path?